### PR TITLE
Unify typed VMVX entry points into untyped functions.

### PR DESCRIPTION
**Pull request by bjacob**
Now that the actual ukernel entry points are untyped, do we still need the VMVX module entry points to be typed?

At least we *can* make them untyped now, and this PR shows it's a bit of a simplification.

In the past we had discussed a few potential benefits of having typed entry points even if they internally all go through the same untyped internal entry point:
 - It could allow to implement target feature support discovery ("does this target have a ukernel for FP16 mmt4d?") as symbol lookup. But thinking about it: isn't the set of supported types just one special angle here, won't we need to perform discovery on more general dimensions ("does this target support this combination of ukernel flags") for which we will need to lift the discovery into proper code anyway (e.g. for each ukernel, have a companion entry point for feature discovery) ?
 - It keeps it an impl detail that the impl is untyped. I would say that's not very important -- worst case we are talking about one well-predicted branch and one function call.

### CODE REVIEW

**Comment by benvanik** [comment]
I think we could make this simplification - it'll keep the binaries smaller too which is nice. The versioning thing is still going to be important at some point especially with datatypes - if there's no f16 ukernel because the hardware doesn't support it but the vmfb uses it we have two options: ship emulated f16 everywhere that needs it in the runtime (prohibitive as we get into weird f8 data types and such and other weird things) or have the compiler emit the fallback path that may widen/use f32/etc). We can cross that bridge when we come to it - there's lots of ways to navigate that (device queries, optional imports, and executable constants would let us branch around instead of needing the native vm import fallbacks and such).

**Comment by bjacob** [comment]
> We can cross that bridge when we come to it - there's lots of ways to navigate that (device queries, optional imports, and executable constants would let us branch around instead of needing the native vm import fallbacks and such).

Right, these were my thoughts too. We will see what concrete examples the future brings, and maybe it will tell us that we need an explicit capability query function, not necessarily something that's a good fit to do as a weak symbol.

**Review by benvanik — APPROVED** [review]

File: runtime/src/iree/modules/vmvx/module.c [hunk 1]
```diff
@@ -686,19 +689,27 @@ IREE_VMVX_ABI_FIXED_STRUCT(unpack, rIIIrIIIIIIIIi, {
...
   IREE_TRACE_ZONE_BEGIN(z0);
   iree_host_size_t out_tile_size = args->in_size2 * args->in_size3;
+  int elem_size = 0;
+  switch (args->flags & IREE_UK_FLAG_UNPACK_TYPE_MASK) {
+    case IREE_UK_FLAG_UNPACK_TYPE_F32F32:
+    case IREE_UK_FLAG_UNPACK_TYPE_I32I32:
+      elem_size = 4;
+      break;
+    default:
+      return iree_make_status(IREE_STATUS_INVALID_ARGUMENT, "unhandled flags");
```
Comment (benvanik) [inline runtime/src/iree/modules/vmvx/module.c:701]: you need to end the trace zone here (or move the begin after this, which is simpler)

**Review by benvanik — CHANGES_REQUESTED** [review]

File: runtime/src/iree/modules/vmvx/module.c (hunk 1 above)
Comment (benvanik) [reply runtime/src/iree/modules/vmvx/module.c:701]: (here and above)

**Review by benvanik — APPROVED** [review]

**Pull request merged**